lib/luadata.c

```diff
@@ -27,11 +27,17 @@ static int luadata_lnew(lua_State *L);
 
 LUNATIK_PRIVATECHECKER(luadata_check, luadata_t *);
 
+static inline void luadata_checkarg(lua_State *L, int arg, size_t size, lua_Integer offset, lua_Integer length)
+{
+	int cond = offset >= 0 && length > 0 && offset + length <= size;
+	luaL_argcheck(L, cond, arg, "out of bounds");
+}
+
 static inline luadata_t *luadata_checkdata(lua_State *L, lua_Integer *offset, lua_Integer length)
 {
 	luadata_t* data = luadata_check(L, 1);
 	*offset = luaL_checkinteger(L, 2);
-	luaL_argcheck(L, *offset >= 0 && length > 0 && *offset + length <= data->size, 2, "out of bounds");
+	luadata_checkarg(L, 2, data->size, *offset, length);
```

Comment:
I'm not sure we should maintain `luadata_checkdata()` only to get `offset`.. it sounds like a unnecessary indirection now we have `luadata_check()` and `luadata_checkbounds()`.. mixed feelings though.. what do you think?

@sheharyaar would like to hear your thoughts as well..

Reply:
I added it to a different commit in case we decide not to make this change.